### ISSUE TITLE
analytics-zoo

### ISSUE THREAD

**Issue by junxnone** [junxnone/tio#941]
# Reference
- [analytics-zoo - intel](https://github.com/intel-analytics/analytics-zoo)
- [docs](https://analytics-zoo.github.io/master/#)
- [Image segmentation unet orca code](https://github.com/intel-analytics/analytics-zoo/tree/master/pyzoo/zoo/examples/orca/learn/tf/image_segmentation)
- [az-orca-tf2 example](https://github.com/intel-analytics/analytics-zoo/blob/master/docs/docs/colab-notebook/orca/quickstart/tf2_keras_lenet_mnist.ipynb)

# Brief
- Distributed TensorFlow/Keras/Pytorch Training and Inference on Apache Spark
- Scale AI model  to large clusters
- Deploy AI pipeline to clusters
- AutoML(support for time series)

![arch](https://raw.githubusercontent.com/intel-analytics/analytics-zoo/master/docs/docs/Image/blockdiagram.jpg)

# Orca
- Scaling out single node Python AI pipeline across large clusters
- Support Framework - Sklearn/Tensorflow/Pytorch/Keras/MXNet/Horovod
- Support Input - Pandas/Numpy/PIL/Tensoflow Dataset/Pytorch DataLoader

> `@20210322` 
> - tf1.x 支持不好，不支持 `callback/tensorboard`, 对 loss/metrics 支持不好
> - tf2.x 支持 `callback/tensorboard`，基本无痛移植

# Term
- BigDL : 分布式深度学习库, 应用可以直接在 `Apache Spark` 上使用
- Apache Spark: 开源集群计算框架 `Spark SQL`/`MLlib`/`GraphX`/`Structured Streaming`
  - Spark ML pipeline
- Apache Flink :  开源流处理框架
- Ray
  - RayOnSpark
- Apache Hadoop YARN -`Yet Another Resource Negotiator` - 资源管理系统
  - Apache Hadoop - 分布式系统基础框架
- K8s - kubernetes - 容器编排引擎
- Databricks
- Google Dataproc
- Cluster Serving
- Scala
- TFPark
